apps/cow-fi/hooks/useLazyLoadImages.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useRef, useCallback } from 'react'
+
+// Shared IntersectionObserver instance
+let observer: IntersectionObserver | null = null
+
+const LAZY_LOADING_CONFIG = {
+  rootMargin: '25px',
+  placeholderColor: '#f0f0f0',
+  fadeInDuration: '0.3s',
+  minHeight: '100px',
+}
+
+// Utility function to generate placeholder src
+const getPlaceholderSrc = (placeholderColor: string): string => {
+  const encodedColor = encodeURIComponent(placeholderColor)
+  return `data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' width='100%' height='100%' viewBox='0 0 1 1' preserveAspectRatio='none'><rect width='1' height='1' fill='${encodedColor}' /></svg>`
+}
+
+// Initialize a shared IntersectionObserver
+const initObserver = (onIntersect: (entry: IntersectionObserverEntry) => void) => {
+  if (observer) return
+
+  observer = new IntersectionObserver(
+    (entries) => {
+      entries.forEach(onIntersect)
+    },
+    { rootMargin: LAZY_LOADING_CONFIG.rootMargin },
+  )
+}
+
+export function useLazyLoadImages() {
```

Comment:
Can we just use [native lazy-loading](https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading) instead?

Reply:
Or https://github.com/Aljullu/react-lazy-load-image-component

Reply:
The issue was that content was being loaded from markdown/cms and  we need to deal what that content